libtsuba/include/tsuba/GR.h

```diff
@@ -0,0 +1,38 @@
+#ifndef KATANA_LIBTSUBA_TSUBA_GR_H_
+#define KATANA_LIBTSUBA_TSUBA_GR_H_
+
+#include <cstdint>
+
+namespace tsuba {
+
+/// The file format for RDG topologies is CSR (Compressed Sparse Row). These
+/// files used to have the file extension .gr (a name tradition continued here)
+/// The structs in this file describe how these GR files are laid out
+
+/// The metadata block at the head of every CSR file
+struct GRHeader {
+  uint64_t version{0};
+  uint64_t edge_type_size{0};
+  uint64_t num_nodes{0};
+  uint64_t num_edges{0};
+};
+
+/// The header and out index array of every CSR file the length of out_indexes
+/// depends on the number of nodes.
+struct GRPrefix {
+  GRHeader header;
+  uint64_t out_indexes[];  // NOLINT needed for layout
+};
+
+constexpr uint64_t
+GRFileSize(const GRHeader header) {
+  uint64_t edge_size =
+      header.version == 1 ? sizeof(uint32_t) : sizeof(uint64_t);
+  return sizeof(header) + ((header.num_nodes - 1) * sizeof(uint64_t)) +
+         (header.num_edges * edge_size) +
```

Comment:
iirc, the gr files have padding as well: [](https://github.com/IntelligentSoftwareSystems/Galois/blob/master/libgalois/include/galois/graphs/LC_CSR_Graph.h#L925)

Reply:
oh! nice catch; luckily since I never read the edge values the padding never matters; I'll update this to be more correct though.